### DISCUSSION TITLE
(Update) Bluetooth Mouse and Keyboard won't work on Dreamcast

### DISCUSSION THREAD

**Discussion by williamdsw** [darthcloud/BlueRetro#842] [Q&A]
Hi, how are you doing?
First, thank you for your work.
I do have some BlueRetro adapters and they work great.
Second, I'm trying to connect a bluetooth mouse and a bluetooth keyboard to my dreamcast, and trying to use on Dreamcast Web Browser.
I've set the mouse on port 1 with default mouse settings, and the keyboard on port 2 with default gamepad/keyboard settings.
Although they pair and appear connected on device list of the browser, I got no input response for both, while they work perfectly on PC.
I'm using the DC adapter and 1.9 firmware.
Also I'm attaching both pcapng files for the mouse & keyboard.
Thank you for your time.
[Keyboard & Mouse pcapng.zip](https://github.com/darthcloud/BlueRetro/files/13519901/Keyboard.Mouse.pcapng.zip)

**Comment by williamdsw** [comment]
Update: The mouse is now working after updating to the 1.9.1 hw1 firmware.
But the keyboard still doesn't work.

**Reply by darthcloud** [reply]
I'll try to look at the KB trace soon.

**Reply by darthcloud** [reply]
I did look at the KB trace finally and BlueRetro is able to decode the HID report properly.

The problem is likely either at the connection phase or at the attribute discovery phase.

I can't debug neither using the traces so I would need to get my hand on that device to debug further.

Where does that KB come from?

**Comment by williamdsw** [comment]
Another Update: I bought a new keyboard and mouse kit from AliExpress, and the new keyboard with US layout works with BlueRetro on Dreamcast (but the mouse from this kit doesn't work).
I tested with the The Typing of the Dead (video), and only a minority of keys aren't mapped correctly.
https://www.youtube.com/watch?v=YVkGMkDJyjM
